sympy/sets/fancysets.py

```diff
@@ -443,3 +444,126 @@ def _sup(self):
     @property
     def _boundary(self):
         return self
+
+
+def as_r_theta(z):
+    """
+    Converts a given Complex Number `z` to polar coordinates.
+
+    Examples
+    ========
+
+    >>> from sympy.sets.fancysets import as_r_theta
+    >>> from sympy import I
+    >>> z = 1 + 2*I
+    >>> r, theta = as_r_theta(z)
+    >>> r
+    sqrt(5)
+    >>> theta
+    pi*atan(2)
+
+    """
+    from sympy import sqrt, atan
+
+    real, imag = z.as_real_imag()
+    r, theta = sqrt(real**2 + imag**2), atan(imag/real)
+    theta = theta % 2*S.Pi
+
+    return r, theta
+
+
+class ComplexPlane(Set):
+    r"""
+    Represents the Set of all Complex Numbers. It can represent a
+    region of Complex Plane in both the standard forms Polar and
+    Rectangular coordinates.
+
+    * Polar Form
+      Input is in the form of the intervals of r and theta.
+
+    Z = {z in C | z = r*[cos(theta) + I*sin(theta)], r in [r], theta in [theta]}
+
+    * Rectangular Form
+      Input is in the form of the interval of x and iota times the
+      interval of y coordinates of the Complex numbers.
+
+                 z = x + I*y
+
+    Z = {z in C | z = x + I*y, x in [Re(z)], y in [Im(z)]}
+
+    Examples
+    ========
+
+    >>> from sympy.sets.fancysets import ComplexPlane
+    >>> from sympy.sets import Interval
+    >>> from sympy import S, I
+    >>> a = Interval(2, 3)
+    >>> b = Interval(4, 6)
+    >>> c1 = ComplexPlane(a, b)  # Rectangular Form
+    >>> c1
+    ComplexPlane(Lambda((x, y), x + I*y), [2, 3] x [4, 6])
+
+    * c1 represents the rectangular region in complex plane
+      surrounded by the coordinates (2, 4), (3, 4), (3, 6) and
+      (2, 6), of the four vertices.
+
+    >>> 2.5 + 4.5*I in c1
+    True
+    >>> 2.5 + 6.5*I in c1
+    False
+
+    >>> r = Interval(0, 1)
+    >>> theta = Interval(0, 2*S.Pi)
+    >>> c2 = ComplexPlane(r, theta, polar=True)  # Polar Form
+    >>> c2  # unit Disk
+    ComplexPlane(Lambda((r, theta), r*(I*sin(theta) + cos(theta))),
+                 [0, 1] x [0, 2*pi])
+
+    * c2 represents the region in complex plane inside the
+      Unit Disk centered at the origin.
+
+    >>> 0.5 + 0.5*I in c2
+    True
+    >>> 1 + 2*I in c2
+    False
+
+    See Also
+    ========
+
+    Reals
+
+    """
+    is_ComplexPlane = True
+
+    def __new__(cls, a_interval, b_interval, polar=False):
+        from sympy import symbols
+
+        x, y, r, theta = symbols('x, y, r, theta')
+        I = S.ImaginaryUnit
+
+        a = a_interval
+        b = b_interval
+
+        # Rectangular Form
+        if polar is False:
+            return ImageSet.__new__(cls, Lambda((x, y), x + I*y), a*b)
+
+        # Polar Form
+        elif polar is True:
+            from sympy import cos, sin
+            return ImageSet.__new__(cls, Lambda((r, theta),
+                                    r*(cos(theta) + I*sin(theta))), a*b)
+
+    def _contains(self, other):
```

Comment:
As @moorepants suggested in https://github.com/sympy/sympy/pull/9438#discussion_r31336944, it is recommended that you write the tests before you write the code for a particular feature, it will the expectations from the code clear and it will also make it easier for us to test if your code is correct or not.

I saw that you have written doctests which partially solves the purpose but unit tests gives a clearer picture of the  behaviour of code.